LICENSE.md

```diff
@@ -73,19 +79,19 @@ The User may examine the Mod's source code.
 The Author reserves the right to distribute the Mod.  
 The Author may designate a person or organization as a Distributor.  
 A Distributor is granted the right to distribute the Mod.  
-The Author may require that a Distributor provides a direct link to the Codebase.  
+The Author may require that a Distributor provides a direct link to the Mod.  
 The User may not distribute the Mod without being granted distribution rights from the Author.  
-The Author may revoke distribution rights from any Distributor.  
+The Author may revoke distribution rights from any Distributor without prior notice.  
 
 6. Right to contribute
 --------
 
-The User may "fork" the Codebase.  
+The User may fork the Mod or any Software associated.  
 The User may compile the Codebase.  
 The User may submit contributions to the Author for inclusion in the Mod.  
 The User grants the Author all rights to any contribution.  
-The User renounces all rights to their contribution except as specified under this license.  
-The User retains the right to re-use code they've written.  
+The User renounces all rights to their contribution except as specified otherwise by the Author.  
+The User retains the right to use any contrbutions they've made.  
```

Comment:
No. Absolutely not. Contributed code is code that the contributor has GIVEN to the author. They retain the right to use it, but they are completely giving up their ownership of that specific bit of code.

A clause like this one is what allowed a single contributor to eviscerated the Bukkit project, and I **will not** allow language like this into the license.